pkg/client/multi.go

```diff
@@ -511,6 +513,43 @@ func (m *MultiClient) Info(ctx context.Context) ([]apiv1.Info, error) {
 	})
 }
 
+func (m *MultiClient) EventStream(ctx context.Context, opts *EventStreamOptions) (<-chan apiv1.Event, error) {
+	clients, err := m.Factory.List(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	all := make(chan apiv1.Event)
+	g, gCtx := errgroup.WithContext(ctx)
+	for _, client := range clients {
+		c := client
+		g.Go(func() error {
+			stream, err := c.EventStream(gCtx, opts)
+			if err != nil {
+				return err
+			}
+
+			for e := range stream {
+				select {
+				case <-gCtx.Done():
+					return gCtx.Err()
+				case all <- e:
+				}
+			}
+
+			return nil
+		})
+	}
+	go func() {
+		defer close(all)
+		if err := g.Wait(); err != nil {
+			logrus.Errorf("Error streaming events: %s", err.Error())
```

Comment:
Should I be returning errors like these over the channel instead of logging? (would require a small refactor to a broader channel type)

Reply:
This may be fine?

If an error occurs here, then the error will be printed and then the channel will be closed. Closing the channel will indicate to any "watchers" that nothing else is coming.

Is this OK?

Reply:
yeah, I think that part is okay -- I'm mostly concerned about how the error looks to users -- this skips any formatting that happens downstream